### PR TITLE
Impute zeroes variable collections

### PR DESCRIPTION
**Pull request by d-callan**
oops. forgot about collections when i built the impute zeroes helper..

### CODE REVIEW

File: R/methods-Megastudy.R [hunk 1]
```diff
@@ -22,6 +22,26 @@ setMethod('getVariableSpec', signature('ANY'), function(object) {
...
 })
 
+# this might be unexpected behavior. should there be a param to choose between the collection and its member specs?
+#' @export
+setMethod('getVariableSpec', signature('VariableMetadata'), function(object, getCollectionMemberVarSpecs = c(TRUE, FALSE)) {
+  getCollectionMemberVarSpecs <- veupathUtils::matchArg(getCollectionMemberVarSpecs)
+  varSpecs <- list(object@variableSpec)
+
+  if (object@isCollection && getCollectionMemberVarSpecs) {
+    varSpecs <- as.list(object@members)
```
Comment (asizemore) [inline R/methods-Megastudy.R:32]: do we need to also get the var specs of the members? I dont remember what the members slot looks like

**Review by asizemore — COMMENTED** [review]
Lots of questions - maybe the three of us could meet and go through it? I'm having a hard time keeping it all straight!

File: R/methods-Megastudy.R (hunk 1 above)
Comment (d-callan) [reply R/methods-Megastudy.R:32]: the members slot is a VariableSpecList, im converting that here to a standard list of VariableSpecs

File: tests/testthat/test-class-Megastudy.R [hunk 2]
```diff
@@ -544,4 +548,42 @@ test_that("imputeZeroes method is sane", {
...
+    new("VariableMetadata",
+      variableClass = new("VariableClass", value = 'native'),
+      variableSpec = new("VariableSpec", variableId = 'specimen_count', entityId = 'sample'),
+      plotReference = new("PlotReference", value = 'yAxis'),
+      dataType = new("DataType", value = 'NUMBER'),
+      dataShape = new("DataShape", value = 'CONTINUOUS'))
+  ))
+
+  imputedDT <- getDTWithImputedZeroes(m, variables)
+  # result has the columns needed to build a plot, based on variables AND the correct number of rows/ zeroes
```
[the remark this answers is not in the document]
Comment (d-callan) [reply tests/testthat/test-class-Megastudy.R:584]: yea its going to plot.data, so im leaving it for now. i know i should do better but i dont have the time right now.

File: tests/testthat/test-class-Megastudy.R [hunk 3]
```diff
@@ -544,4 +548,42 @@ test_that("imputeZeroes method is sane", {
   ))
 
   expect_error(getDTWithImputedZeroes(m, variables))
+
+  # variable collection exists in plot
+  pathogenVariableCollectionVocabs <- StudySpecificVocabulariesByVariable(S4Vectors::SimpleList(StudySpecificVocabulary(studyIdColumnName='study.id', study='a', variableSpec=VariableSpec(entityId='assay',variableId='pathogen_presence_variable_collection'), vocabulary=c('Yes','No')),
```
[the remark this answers is not in the document]
Comment (d-callan) [reply tests/testthat/test-class-Megastudy.R:553]: can they differ? these are collections after all... maybe i should err?

[the remark this answers is not in the document]
Comment (d-callan) [reply tests/testthat/test-class-Megastudy.R:553]: or maybe i guess one study could have 'maybe' and not another? my head hurts..

File: R/methods-VariableMetadata.R [hunk 4]
```diff
@@ -628,13 +633,13 @@ setMethod("findColNamesByPredicate", signature("VariableMetadataList"), function
...
 setGeneric("findVariableMetadataFromVariableSpec", 
-  function(variables, object) standardGeneric("findVariableMetadataFromVariableSpec"),
+  function(variables, object, ...) standardGeneric("findVariableMetadataFromVariableSpec"),
   signature = c("variables","object")
 )
 
 #' @export
 setMethod("findVariableMetadataFromVariableSpec", signature("VariableMetadataList", "VariableSpecList"), function(variables, object) {
-  variableSpecs <- lapply(as.list(variables), veupathUtils::getVariableSpec)
+  variableSpecs <- unlist(lapply(as.list(variables), veupathUtils::getVariableSpec, FALSE))
```
[the remark this answers is not in the document]
Comment (d-callan) [reply R/methods-VariableMetadata.R:642]: the collection members dont have full metadata to return.

File: tests/testthat/test-class-Megastudy.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (d-callan) [reply tests/testthat/test-class-Megastudy.R:553]: this feels like a rabbit hole. im going to leave it until we actually see a case..

File: tests/testthat/test-class-Megastudy.R [hunk 5]
```diff
@@ -544,4 +548,42 @@ test_that("imputeZeroes method is sane", {
...
+      variableSpec = new("VariableSpec", variableId = 'pathogen_presence_variable_collection', entityId = 'assay'),
+      plotReference = new("PlotReference", value = 'overlay'),
+      dataType = new("DataType", value = 'STRING'),
+      dataShape = new("DataShape", value = 'CATEGORICAL'),
+      isCollection = TRUE,
+      weightingVariableSpec = VariableSpec(variableId='specimen_count',entityId='sample'),
+      hasStudyDependentVocabulary = TRUE,
+      members = VariableSpecList(S4Vectors::SimpleList(VariableSpec(variableId='pathogen_presence', entityId='assay'),
+                                                       VariableSpec(variableId='pathogen2_presence', entityId='assay'),
+                                                       VariableSpec(variableId='pathogen3_presence', entityId='assay')))
```
Comment (d-callan) [inline tests/testthat/test-class-Megastudy.R:573]: i think i need a test where i have a collection, but the OTHER var is the one w the special vocab

**Pull request merged**